### PR TITLE
feat: add trimspace to sanitize yaml fields

### PR DESCRIPTION
**Pull request by aramase**
**What this PR does / why we need it**:
- Add `strings.TrimSpace` for all fields obtained as string from manifests
- Adds positive test cases for `GetSecretData`

**Which issue(s) this PR fixes** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/util/secretutil/secret.go [hunk 1]
```diff
@@ -157,7 +158,7 @@ func GetSecretData(secretObjData []*v1alpha1.SecretObjectData, secretType corev1
 		if len(data.Key) == 0 {
 			return datamap, fmt.Errorf("key in secretObjects.data is empty")
 		}
-		file, ok := files[data.ObjectName]
+		file, ok := files[strings.TrimSpace(data.ObjectName)]
```
Comment (ritazh) [inline pkg/util/secretutil/secret.go:161]: Do we also need to trim space for `data.Key`?

Comment (aramase) [reply pkg/util/secretutil/secret.go]: Updated the PR!

File: pkg/util/secretutil/secret.go [hunk 2]
```diff
@@ -151,27 +152,30 @@ func ValidateSecretObject(secretObj v1alpha1.SecretObject) error {
...
+
+		if len(objectName) == 0 {
 			return datamap, fmt.Errorf("object name in secretObjects.data")
 		}
-		if len(data.Key) == 0 {
+		if len(dataKey) == 0 {
 			return datamap, fmt.Errorf("key in secretObjects.data is empty")
 		}
-		file, ok := files[data.ObjectName]
+		file, ok := files[strings.TrimSpace(objectName)]
```
Comment (ritazh) [inline pkg/util/secretutil/secret.go:164]: why do you need to trim again here?

Comment (aramase) [reply pkg/util/secretutil/secret.go]: s/`data.ObjectName`/`objectName` 🤦 

Updated it!

**Pull request merged**